extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/KryoFeatureDataAdapter.java

```diff
@@ -128,13 +131,32 @@ public AdapterPersistenceEncoding encode(
 		final byte[] typeNameBytes = StringUtils.stringToBinary(typeName);
 		final byte[] visibilityManagementClassNameBytes = StringUtils.stringToBinary(fieldVisibilityManagement.getClass().getCanonicalName());
 		final byte[] encodedTypeBytes = StringUtils.stringToBinary(encodedType);
-		final ByteBuffer buf = ByteBuffer.allocate(encodedTypeBytes.length + typeNameBytes.length + visibilityManagementClassNameBytes.length + adapterId.getBytes().length + 16);
+		final ByteBuffer buf = ByteBuffer.allocate(encodedTypeBytes.length + typeNameBytes.length + visibilityManagementClassNameBytes.length + adapterId.getBytes().length + 20);
```

Comment:
is this sufficient, looks to me that we added 4 bytes , when I see (8 + attrbytes.length)